xen/drivers/passthrough/iommu.c

```diff
@@ -249,22 +249,39 @@ void iommu_domain_destroy(struct domain *d)
     arch_iommu_domain_destroy(d);
 }
 
-int iommu_map_page(struct domain *d, unsigned long gfn, unsigned long mfn,
-                   unsigned int flags)
+int iommu_map_pages(struct domain *d, unsigned long gfn, unsigned long mfn,
+                    unsigned long page_count, unsigned int flags)
 {
     const struct domain_iommu *hd = dom_iommu(d);
-    int rc;
+    int rc = 0;
```

Comment:
Something like this:
 if ( hd->platform_ops->unmap_pages )
        rc = hd->platform_ops->unmap_pages(d, gfn, page_count);
    else
    {
        for ( i = 0; i < page_count; i++, gfn++ )
        {
            ret = hd->platform_ops->unmap_page(d, gfn);
            if ( likely(!rc) )
                rc = ret;
        }
    }